src/EventStore.Core/Util/MD5Hash.cs

```diff
@@ -59,14 +59,12 @@ public static async ValueTask ContinuousHashFor(HashAlgorithm md5, Stream s, int
 
 		var buffer = ArrayPool<byte>.Shared.Rent(4096);
 		try {
-			long toRead = count;
-			while (toRead > 0) {
-				int read = await s.ReadAsync(buffer.AsMemory(0, (int)Math.Min(toRead, buffer.Length)), token);
-				if (read is 0)
+			for (int bytesRead; count > 0L; count -= bytesRead) {
+				bytesRead = await s.ReadAsync(buffer.AsMemory(0, (int)Math.Min(count, buffer.Length)), token);
```

Comment:
```suggestion
				bytesRead = await s.ReadAsync(buffer.AsMemory(0, (int)Math.Min(count, 4096)), token);
```

per our earlier discussion, i agree that `ArrayPool` guarantees that an array of length 4096 is returned here as it's a power of 2, but it's an implementation detail, isn't it ? (i don't see it mentioned in the API docs)

Reply:
Also, if there was a buffer larger than 4k rented previously, the pool returns that buffer. It can be 8k, 16k, or any other size. Larger buffer could improve I/O performance, because we can read more bytes at the cost of single system call, which is more beneficial.